File: 0134-gas-station/0134-gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        totalGas,totalCost = sum(gas), sum(cost)
        curGass,start = 0,0

        if totalGas < totalCost:
            return -1

        for i in range(len(gas)):
            curGass += gas[i] - cost[i]

            if curGass < 0 :
                start = i+1
                curGass = 0
        
        return -1 if totalGas < totalCost else start


        

File: 0134-gas-station/0134-gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int totalGas = 0, totalCost = 0;
        int currentGas = 0, start = 0;

        for (int i = 0; i < gas.size(); i++) {
            totalGas += gas[i];
            totalCost += cost[i];
            currentGas += (gas[i] - cost[i]);

            if (currentGas < 0) {
                start = i + 1;
                currentGas = 0;
            }
        }

        return totalGas < totalCost ? -1 : start;
    }
};